src/main/java/com/google/cloudassets/discovery/assetobjects/AssetObject.java

```diff
@@ -75,31 +74,23 @@ public B setKind(Object kind) {
         Set the name field of this object with the provided string and return its specific Builder.
         */
         public B setName(Object name) {
-            specificObjectClass.name = (String) name;
+            specificObjectClass.name = convertObjectToString(name);
```

Comment:
I know this is not part of this PR, but I find the name "convertToString" a bit misleading, because it doesn't really convert but rather tries to cast to string with fallback to null. Maybe we can change this to "castToString" or similar? (Can do in another PR)

Reply:
You are right, I will indeed add it to the next PR as I already started adding new commits for the new PR in the same branch (as a mistake)